### PR TITLE
feat: sanitise personalisation field for emails

### PR DESCRIPTION
**Pull request by joybytes**
## Summary
Added optional `sanitise_content_for` for send-email  and `sanitised_content `on the response. Tests, changelog, and version bump included.

Ran full integration tests and they pass.

## Checklist
- [x] I’ve used the pull request template
- [x] I’ve written unit tests for these changes
- [x] I’ve updated the documentation in
  - [x] [notifications-tech-docs repository](https://github.com/alphagov/notifications-tech-docs/blob/main/source/documentation/client_docs/_ruby.md)
  - [x] `CHANGELOG.md`
- [x] I’ve bumped the version number (in `lib/notifications/client/version.rb`)

### CODE REVIEW

File: spec/notifications/client/email_notification_spec.rb [hunk 1]
```diff
@@ -47,5 +48,59 @@
...
+      }
+
+      it "serialises sanitise_content_for as an empty JSON array" do
+        expect(WebMock).to have_requested(:post, "https://#{uri.host}:#{uri.port}/v2/notifications/email").with { |req|
+          JSON.parse(req.body)["sanitise_content_for"] == []
+        }
+      end
+    end
+
+    context "when the API response omits sanitised_content" do
```
Comment (kr8n3r) [inline spec/notifications/client/email_notification_spec.rb]: I'm sure you're checking correctly for `nil` which means variable is returned but is empty. API always returns `santised_content`

File: bin/test_client.rb [hunk 2]
```diff
@@ -186,6 +187,9 @@ def test_notification_response_data_type(notification, message_type)
 
   if message_type == 'email'
     hash_key_should_not_be_nil(expected_fields_in_email_content, notification.send('content'), 'send_' + message_type + '.content')
+    unless notification.sanitised_content.is_a?(Hash)
```
Comment (kr8n3r) [inline bin/test_client.rb]: is `Hash` too lax? API expects an array of strings

File: spec/notifications/client/email_notification_spec.rb (hunk 1 above)
Comment (joybytes) [reply spec/notifications/client/email_notification_spec.rb]: thanks
I’ve removed that example and added a check that sanitised_content is always an object (a Hash in Ruby), which can be empty.

File: bin/test_client.rb (hunk 2 above)
Comment (joybytes) [reply bin/test_client.rb]: sanitise_content_for on the request expects array of strings
sanitised_content on the response is an object

I've also tried to improve the testing to include check for empty object and also the "sanitised" and "unsanitised" keys on `sanitised_content_matches_email_send_api_response`

**Review by kr8n3r — APPROVED** [review]
this looks good to me
